### PR TITLE
[experimental][rocm] Added tracing for rocm backend.

### PR DESCRIPTION
**Pull request by raikonenfnu**

### CODE REVIEW

File: experimental/rocm/direct_command_buffer.c [hunk 1]
```diff
@@ -338,32 +341,48 @@ static iree_status_t iree_hal_rocm_direct_command_buffer_dispatch(
...
-  IREE_RETURN_IF_ERROR(iree_hal_rocm_native_executable_block_size(
-      executable, entry_point, &block_size_x, &block_size_y, &block_size_z));
-  hipFunction_t func =
-      iree_hal_rocm_native_executable_for_entry_point(executable, entry_point);
+  // int32_t block_size_x, block_size_y, block_size_z;
+  // IREE_RETURN_IF_ERROR(iree_hal_rocm_native_executable_block_size(
+  //     executable, entry_point, &block_size_x, &block_size_y, &block_size_z));
+  // hipFunction_t func =
+  //     iree_hal_rocm_native_executable_for_entry_point(executable,
+  //     entry_point);
```
Comment (benvanik) [inline experimental/rocm/direct_command_buffer.c]: dead code

**Review by benvanik — CHANGES_REQUESTED** [review]
lgtm besides 1 bit of dead code!

File: experimental/rocm/direct_command_buffer.c (hunk 1 above)
Comment (raikonenfnu) [reply experimental/rocm/direct_command_buffer.c]: Done!

**Comment by raikonenfnu** [comment]
> lgtm besides 1 bit of dead code!

Thanks Ben! :)

**Review by benvanik — APPROVED** [review]

**Pull request merged**